libs/api/database/src/lib/content/stores/tags.store.ts

```diff
@@ -22,6 +22,25 @@ export class TagsStore {
         return this.tags.find({ kind: kind }).sort({ name: 1 });
     }
 
+    /**
+     * Returns all tags of a specified kind, sorted by parent (none first) then alphabetically.
+     * @param kind
+     */
+    async fetchTagsSortedByParent(kind: TagKind): Promise<TagsDocument[]> {
+        return this.tags.find({ kind: kind }).sort({ parent: 1, name: 1 });
```

Comment:
You might be able to achieve something similar with Mongo's `graphLookup()`; that can give you a flat list of _all_ tags, but it'll preserve their parent/child relationships. If you want them as a `TagsTree`, you'll have to reconstruct it it a bit, but it might be easier.